src/main/java/com/minecolonies/coremod/client/gui/WindowHutBuilder.java

```diff
@@ -35,16 +65,46 @@ public WindowHutBuilder(final BuildingBuilder.View building)
         super(building, Constants.MOD_ID + HUT_BUILDER_RESOURCE_SUFFIX);
         this.builder = building;
         pullResourcesFromHut();
+        registerButton(RESOURCE_ADD, this::transferItems);
     }
 
     /**
      * Retrieve resources from the building to display in GUI.
      */
     private void pullResourcesFromHut()
     {
-        if (builder.getColony().getBuilding(builder.getID()) != null)
+        final AbstractBuilding.View newView = builder.getColony().getBuilding(builder.getID());
+        if (newView != null && newView instanceof BuildingBuilder.View)
         {
-            resources = builder.getNeededResources();
+            final BuildingBuilder.View updatedView = (BuildingBuilder.View) newView;
+            final InventoryPlayer inventory = this.mc.player.inventory;
+ 
+            resources.clear();
+            resources.addAll(updatedView.getResources().values());
+            for (int i =0; i<resources.size();i++)
+            {
+                final BuildingBuilder.BuildingBuilderResource resource = resources.get(i);
+                final Item item = resource.getItemStack().getItem();
+                resource.setPlayerAmount(InventoryUtils.getItemCountInInventory(inventory, item, resource.getItemStack().getItemDamage()));
+            }
+
+            Collections.sort(resources, new Comparator<BuildingBuilder.BuildingBuilderResource>() 
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1604)